csharp/Item.cs

```diff
@@ -1,14 +1,69 @@
-﻿namespace csharp
+﻿using System.Collections.Generic;
+using NDesk.Options;
+
+namespace csharp
 {
     public class Item
     {
         public string Name { get; set; }
         public int SellIn { get; set; }
         public int Quality { get; set; }
+        public string Type { get; set; }
+        public bool Perishable { get; private set; }
+        public int Change { get; set; }
+        public Dictionary<int, int> MileStones { get; private set; }
+        
+        public void AffixAttributes()
```

Comment:
You are changing the input, adding a type -> You should be using the string presented, you can add them to constants
The Milestones is a good touch but trickier to understand at first.

You should also call this method from UpdateQuality method or even in constructor rather than before initialising the class